### PR TITLE
Add authorized filter

### PR DESCRIPTION
**Pull request by cedricgrothues**
This PR adds an `authorized` filter that authorizes users based on the `token` cookie.

### CODE REVIEW

File: server/src/main.rs [hunk 1]
```diff
@@ -44,6 +44,22 @@ async fn main() {
     ))
     .expect("could not create db connection");
 
+    let authorized = with_db(database.clone())
+        .and(warp::cookie("token"))
+        .and_then(|db: db::Pool, token: String| async move {
+            let res = data::author::dsl::author
+                .filter(data::author::dsl::access_token.eq(&token))
+                .execute(&db.get().unwrap());
```
Comment (teymour-aldridge) [inline server/src/main.rs:52]: I reckon you can do this with `first`
```suggestion
                .select(data::author::id)
                .first::<i32>(&db.get().unwrap());
```

File: server/src/main.rs [hunk 2]
```diff
@@ -44,6 +44,22 @@ async fn main() {
...
+
+            if let Ok(num_res) = res {
+                if num_res > 0 {
+                    return Ok(token);
```
Comment (teymour-aldridge) [inline server/src/main.rs:56]: I think it's more helpful to return the user's id rather than the access token.

Comment (cedricgrothues) [reply server/src/main.rs:56]: Sounds good!

File: server/src/main.rs (hunk 1 above)
Comment (cedricgrothues) [reply server/src/main.rs:52]: What is returned here if the `user` doesn't exist?

Comment (teymour-aldridge) [reply server/src/main.rs]: It's an error.

Comment (teymour-aldridge) [reply server/src/main.rs]: https://docs.rs/diesel/1.4.5/diesel/result/enum.Error.html#variant.NotFound

**Pull request merged**